pkg/skuba/actions/node/upgrade/apply.go

```diff
@@ -118,12 +118,14 @@ func Apply(target *deployments.Target) error {
 			if err != nil {
 				return err
 			}
-			err = target.Apply(deployments.KubernetesBaseOSConfiguration{
-				KubeadmVersion:    nodeVersionInfoUpdate.Update.APIServerVersion.String(),
-				KubernetesVersion: nodeVersionInfoUpdate.Current.APIServerVersion.String(),
-			}, "kubernetes.install-intermediate-node-pattern")
-			if err != nil {
-				return err
+			if nodeVersionInfoUpdate.HasMajorOrMinorUpdate() {
```

Comment:
do we also have to put this around the `install-node-pattern` later after `kubeadm.upgrade.apply`? it's going to be introduced by https://github.com/SUSE/skuba/pull/585 

also possibly some other states don't have to be run if we don't have a major or minor update, e.g. restarting services?

Reply:
I prefer to play safe and improve later :)